sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java

```diff
@@ -85,31 +101,46 @@ private static void extractPartnerUrl(ForceUserInfo userInfo) {
         userInfo.setPartnerUrl(userInfo.getUrls().get("partner").replace("{version}", API_VERSION));
     }
 
-    private static boolean isBadTokenError(HttpResponseException e) {
-        return e.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN;
+    private boolean isBadTokenError(HttpResponseException e) {
+        return ((e.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN)
+                &&
+                StringUtils.equalsAnyIgnoreCase(responseContent, BAD_TOKEN_SF_ERROR_CODE) ||
+                StringUtils.equalsAnyIgnoreCase(responseContent, MISSING_TOKEN_SF_ERROR_CODE) ||
+                StringUtils.equalsAnyIgnoreCase(responseContent, WRONG_ORG_SF_ERROR_CODE))
+                ||
+                (e.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND &&
+                        StringUtils.equalsAnyIgnoreCase(responseContent, BAD_ID_SF_ERROR_CODE));
```

Comment:
Please use equalsIgnoreCase() for single constant

Reply:
fixed